### DISCUSSION TITLE
Controller for PC lighting?

### DISCUSSION THREAD

**Discussion by 125millibytes** [PlummersSoftwareLLC/NightDriverStrip#304] [Q&A]
Hi!

Can anybody recommend a controller to put effects on addressable LED strips and fans inside a PC? 
I got 6 internal fans, two LED strips, and an external radiator with 9 fans. The radiator could be fit with its own controller.

Since they will be connected to the PC internally, communication over USB would be perfect. I can make cables to connect two controllers on a single USB2 header on the mainboard.

Thanks!
Daniel

**Comment by davepl** [comment]
I use the M5Stick C Plus for this very project.

**Reply by 125millibytes** [reply]
Hi Dave!

What do you mean by "this very project"? Do you mean you use the M5StickC to develop NightDriver?

I'm looking for a controller to hide somewhere in a PC and connect all the ARGB signal wires. As far as I understand, if the LED strips are WS2812B, I can hook them up to a common ground and +5V from a SATA power plug and just use one GPIO pin from an ESP32 to control each strip/FAN, correct?
I don't really need an LCD and battery. And the M5STick doesn't seem to have the amount of GPIO pins I need to connect 9 strips.

Sorry I don't have much experience with this stuff. Am I missing something? I believe the perfect minimal solution would be a ESP32 board with just USB and lots of GPIO pins. I'll definitely have a look at the other products from M5Stack

Thank you!

**Comment by davepl** [comment]
Yes, there’s a project in there for PC FANs already, though I forget the exact name.  It has functions to draw into the circles, draw at clock positions, draw across circular surfaces, and so on. 

You could use any ESP32 module.  I just like the M5.

In my case I plug it into an internal USB header and power it that way.  You can get power from SATA as well, as you note… likely more.

- Dave

**Comment by 125millibytes** [comment]
OK I'll check if I can get enough power from the USB header directly.

And it might actually be a good idea to get a controller with many features, just to have options while I get started learning this stuff.

But do I understand correctly that each LED strip/fan needs to be connected to one GPIO pin of the ESP32? Right at the start of the Readme it says "It can drive up to 8 channels". Is that a software or hardware limitation? I would need exactly 9 for my 9 external radiator fans.

edit: 
I understand now that the 8 channel limit comes from FastLED because it uses ESP32's [RMT](https://docs.espressif.com/projects/esp-idf/en/v4.4/esp32/api-reference/peripherals/rmt.html). 

AcidAngel_ over at [r/FastLED](https://www.reddit.com/r/FastLED/comments/140gfrf/comment/jmvqj4d/) said:

> esp32 can feed 16 led strips in parallel. Just write these lines.
> define FASTLED_ESP32_I2S true
> include "FastLED.h"

Would that be a possibility in NightDriver?

I also ordered the M5Stick to just have a go and get started with NightDriver and/or FastLED. 

Thanks!
